features/home/providers/claiming-provider.tsx

```diff
@@ -0,0 +1,35 @@
+import {
+  createContext,
+  Dispatch,
+  FC,
+  PropsWithChildren,
+  SetStateAction,
+  useContext,
+  useState,
+} from 'react';
+
+export type ClaimingContextType = {
+  isClaiming: boolean;
+  setIsClaiming: Dispatch<SetStateAction<boolean>>;
+};
+
+export const ClaimingContext = createContext<ClaimingContextType>(
+  {} as ClaimingContextType,
+);
+
+export const ClaimingProvider: FC<PropsWithChildren> = ({ children }) => {
```

Comment:
Do you really need the variable `isClaiming` and this provider now?

Reply:
Not sure, it is used in some places, but not sure if it's actually valuable right now, I've removed a few states on the way, but they were not needed 🤔

Reply:
It is used in a wallet 
<img width="1512" alt="image" src="https://user-images.githubusercontent.com/103929444/225418501-d512fd76-3ebc-437b-a85e-05a4ae28dad6.png">

Reply:
These are the TODOs that I spoke about and which it is desirable to correct if possible

Reply:
Yes, I know. Therefore, this code is marked as TODO. I think it will be useful in the future to think about how to do it better

Reply:
Changes in this PR aren't about anything in these files, if there will be any work related to them or any issues, we will refactor, but this is out of scope for this PR 🤔